worlds/stardew_valley/logic/ability_logic.py

```diff
@@ -13,14 +13,20 @@
 from ..strings.skill_names import Skill, ModSkill
 from ..strings.tool_names import ToolMaterial, Tool
 
+if typing.TYPE_CHECKING:
+    from ..mods.logic.mod_logic import ModLogicMixin
+else:
+    ModLogicMixin = object
+
 
 class AbilityLogicMixin(BaseLogicMixin):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.ability = AbilityLogic(*args, **kwargs)
 
 
-class AbilityLogic(BaseLogic[Union[AbilityLogicMixin, RegionLogicMixin, ReceivedLogicMixin, ToolLogicMixin, SkillLogicMixin, MineLogicMixin, MagicLogicMixin]]):
+class AbilityLogic(BaseLogic[Union[AbilityLogicMixin, RegionLogicMixin, ReceivedLogicMixin, ToolLogicMixin, SkillLogicMixin, MineLogicMixin, MagicLogicMixin,
+ModLogicMixin]]):

```

Comment:
is this the proper indenting? i would have expected
```suggestion
class AbilityLogic(BaseLogic[Union[AbilityLogicMixin, RegionLogicMixin, ReceivedLogicMixin, ToolLogicMixin, SkillLogicMixin, MineLogicMixin, MagicLogicMixin,
                   ModLogicMixin]]):

```
looks like you have it elsewhere though, so if it's valid and consistent i guess it's fine

Reply:
It is valid, but I hate it as well... I plan on eventually replacing all that mixin union stuff by just using the class that wraps all mixins and validating `typing.TYPE_CHECKING`.